Subject: [PATCH] git-instaweb: configure $mimetypes_file if found

If `git --exec-dir`/../../etc/mime.types exists then configure
gitweb to use it for the mime types file.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 git-instaweb.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/git-instaweb.sh b/git-instaweb.sh
index 2fcdf392..27572efd 100755
--- a/git-instaweb.sh
+++ b/git-instaweb.sh
@@ -607,6 +607,12 @@ our \$highlight_bin = "$hlbin";
 \$feature{'highlight'}{'default'} = [1];
 EOF
 	fi
+	etcdir="$(cd "$GIT_EXEC_PATH/../../etc" 2>/dev/null && pwd -P)"
+	if [ -n "$etcdir" -a -d "$etcdir" -a -f "$etcdir/mime.types" ]; then
+	cat >> "$fqgitdir/gitweb/gitweb_config.perl" <<EOF
+our \$mimetypes_file = "$etcdir/mime.types";
+EOF
+	fi
 }
 
 configure_httpd() {
---
